### PR TITLE
Add support for serviceAccount and podSecurity in cp-kafka-connect

### PR DESCRIPTION
**Pull request by batas**
## What changes were proposed in this pull request?

This change enables add serviceAccount and podSecurity to pod.

## How was this patch tested?

I deployed my fork in my cluster on AWS.

### CODE REVIEW

**Comment by yokiworks** [comment]
We really need that as well

**Comment by yokiworks** [comment]
> It looks like @batas hasn't signed our **C**ontributor **L**icense **A**greement, yet.
> 
> > The purpose of a CLA is to ensure that the guardian of a project's outputs has the necessary ownership or grants of rights over all contributions to allow them to distribute under the chosen licence.
> > [Wikipedia](http://en.wikipedia.org/wiki/Contributor_License_Agreement)
> 
> You can read and sign our full Contributor License Agreement [here](http://clabot.confluent.io/cla).
> 
> Once you've signed reply with `[clabot:check]` to prove it.
> 
> Appreciation of efforts,
> 
> clabot

[clabot:check]

**Comment by ServerNinja** [comment]
@batas This is a much needed change. Thank you so much for submitting this. I see the serviceAccount was only added to the cp-kafka-connect chart. Can we consider including serviceAccount templates for the others?

**Comment by gAmUssA** [comment]
@batas kindly provide details on how to validate this (version of Kubernetes, what needs to be configured, etc) Thank you

**Comment by batas** [comment]
I'm using 1.18.9 eks. 
Example configuration:
```  
serviceAccount:
    name: cp-kafka-connect
    annotations:
      eks.amazonaws.com/role-arn: arn:aws:iam::ID:role/ROLE_NAME
```

I can provide this change for all resources, but until this chart hasn't own repo, I don't want to do it.

**Comment by marcusramberg** [comment]
I need to add the same thing for schema registry.

File: charts/cp-kafka-connect/templates/deployment.yaml [hunk 1]
```diff
@@ -33,9 +33,15 @@ spec:
...
     spec:
+      serviceAccountName: {{ include "cp-kafka-connect.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+
       containers:
         {{- if .Values.prometheus.jmx.enabled }}
         - name: prometheus-jmx-exporter
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
```
Comment (marcusramberg) [inline charts/cp-kafka-connect/templates/deployment.yaml:44]: Are you sure the context should be added to the prometheus exporter as well?

**Pull request closed without merging**